### PR TITLE
Update `README`s to document `dep`-based workflow.

### LINKED ISSUE #578: Update geth-patches readme with dep instruction and creating patches
# Problem

After merging `dep` https://github.com/status-im/status-go/pull/551, working with patches changed a bit, however, [README.md](https://github.com/status-im/status-go/blob/develop/geth-patches/README.md) has not been updated.

# Acceptance Criteria

1. Update `Vendor patched version` section,
2. Create a new section which explains how to create a patch. (This is not related to #551 but it looks like it's missing.)
3. Add a link to the [rebasing docs on Notion](https://www.notion.so/status/Rebasing-go-ethereum-82de0bbebe684826b62d81bc688e5e89)

### PR DESCRIPTION
**Pull request by mandrigin**
Updated workflow for upstream rebasing with `dep`.

Closes #578

### CODE REVIEW

File: geth-patches/README.md [hunk 1]
```diff
@@ -9,44 +9,36 @@ We try to minimize number and amount of changes in those patches as much as poss
...
 
 **Note: The process is completely repeatable, so it's safe to remove current `go-ethereum` directory, clone latest upstream version and apply patches from scratch.**
 
-## How to update forked version
-Make sure you have `status-go` in your `$GOPATH/src/github.com/status-im/` first.
+### Using existing fork repo (recommended)
 
-### From very scratch
-Use this method if you're up to nuke forked repo for some reason and apply patches from scratch:
+#### I. In `go-ethereum fork`.
```
Comment (themue) [inline geth-patches/README.md]: Would write

#### I. In our fork at /status-im/go-ethereum

File: geth-patches/README.md [hunk 2]
```diff
@@ -59,25 +51,23 @@ Once patches applied, you might want to inspect changes between current vendored
...
-
-TBD
+4. Push `develop` branch to our remote, rewriting history
+```
+git push -f origin develop
+```
 
-## Manually
-This method should be used only while `dep` tool workflow is not set up.
+#### II. In `status-go` repository
```
Comment (themue) [inline geth-patches/README.md]: Also here status-go is a name, no code. So simply

#### II. In status-go repository

File: geth-patches/README.md [hunk 3]
```diff
@@ -9,44 +9,36 @@ We try to minimize number and amount of changes in those patches as much as poss
...
+ - [`0000-accounts-hd-keys.patch`](./0000-accounts-hd-keys.patch) — adds support for HD extended keys (links/docs?)
+ - [`0002-les-api-status.patch`](./0002-les-api-status.patch) — adds StatusBackend into LES code (need to be inspected, some things can and should be done outside of les code
+ - [`0003-dockerfiles-wnode-swarm.patch`](./0003-dockerfiles-wnode-swarm.patch) — adds Dockerfiles (who uses this?)
+ - [`0004-whisper-notifications.patch`](./0004-whisper-notifications.patch) — adds Whisper notifications (need to be reviewed and documented)
+ - [`0006-latest-cht.patch`](./0006-latest-cht.patch) – updates CHT root hashes, should be updated regularly to keep sync fast, until proper Trusted Checkpoint sync is not implemented as part of LES/2 protocol.
+ - [`0007-README.patch`](./0007-README.patch) — update upstream README.md.
+ - [`0008-tx-pool-nonce.patch`](./0008-tx-pool-nonce.patch) - On GetTransactionCount request with PendingBlockNumber get the nonce from transaction pool
+ - [`0010-geth-17-fix-npe-in-filter-system.patch`](./0010-geth-17-fix-npe-in-filter-system.patch) - Temp patch for 1.7.x to fix a NPE in the filter system
```
Comment (themue) [inline geth-patches/README.md:19]: Not yet sure if "17" is a good abbreviation for version 1.7.x. But that's more a topic about naming schemes to manage different versions.

**Review by themue — APPROVED** [review]
Only minor formal wishes.

File: geth-patches/README.md [hunk 4]
```diff
@@ -9,44 +9,36 @@ We try to minimize number and amount of changes in those patches as much as poss
...
+git branch -D develop
 ```
 
-### Using existing fork repo (recommended)
-
+2. Pull upstream release branch into `develop` branch.
 ```
-# merge upstream release branch into local master
-git pull git@github.com:ethereum/go-ethereum.git release/1.7:master
+git pull git@github.com:ethereum/go-ethereum.git release/1.7:develop
```
Comment (adambabik) [inline geth-patches/README.md]: We should put a comment saying that upstream release branch depends on the current stable `geth` version we use. Currently, it's `1.7.x`.

File: geth-patches/README.md [hunk 5]
```diff
@@ -59,25 +51,23 @@ Once patches applied, you might want to inspect changes between current vendored
...
+4. Push `develop` branch to our remote, rewriting history
+```
+git push -f origin develop
+```
 
-## Manually
-This method should be used only while `dep` tool workflow is not set up.
+#### II. In `status-go` repository
 
+1. Update vendored `go-ethereum` (note that we use upstream's address there, the link to our fork is in `Gopkg.toml`)
```
Comment (adambabik) [inline geth-patches/README.md]: `[...] the link to our fork is in [...]` maybe `[...] we override the download link to our fork address in [...]`

**Review by adambabik — APPROVED** [review]
LGTM! Some copy suggestions.

File: geth-patches/README.md (hunk 3 above)
Comment (mandrigin) [reply geth-patches/README.md:19]: yeah, I agree.
hope this patch will go away soon anyway :)

**Comment by mandrigin** [comment]
Merging before CI passes because there are no code changes.
@adambabik @themue thanks for your feedback, fixed the points you noted.

**Pull request merged**